app/src/main/java/com/duckduckgo/app/bookmarks/ui/BookmarksActivity.kt

```diff
@@ -176,20 +188,33 @@ class BookmarksActivity : DuckDuckGoActivity() {
             this,
         ) {
             when (it) {
-                is BookmarksViewModel.Command.ConfirmDeleteSavedSite -> confirmDeleteSavedSite(it.savedSite)
-                is BookmarksViewModel.Command.OpenSavedSite -> openSavedSite(it.savedSiteUrl)
-                is BookmarksViewModel.Command.ShowEditSavedSite -> showEditSavedSiteDialog(it.savedSite)
-                is BookmarksViewModel.Command.ImportedSavedSites -> showImportedSavedSites(it.importSavedSitesResult)
-                is BookmarksViewModel.Command.ExportedSavedSites -> showExportedSavedSites(it.exportSavedSitesResult)
-                is BookmarksViewModel.Command.OpenBookmarkFolder -> openBookmarkFolder(it.bookmarkFolder)
-                is BookmarksViewModel.Command.ShowEditBookmarkFolder -> editBookmarkFolder(it.bookmarkFolder)
-                is BookmarksViewModel.Command.DeleteBookmarkFolder -> deleteBookmarkFolder(it.bookmarkFolder)
-                is BookmarksViewModel.Command.ConfirmDeleteBookmarkFolder -> confirmDeleteBookmarkFolder(it.bookmarkFolder)
-                is BookmarksViewModel.Command.LaunchBookmarkImport -> launchBookmarkImport()
+                is ConfirmDeleteSavedSite -> confirmDeleteSavedSite(it.savedSite)
+                is OpenSavedSite -> openSavedSite(it.savedSiteUrl)
+                is ShowEditSavedSite -> showEditSavedSiteDialog(it.savedSite)
+                is ImportedSavedSites -> showImportedSavedSites(it.importSavedSitesResult)
+                is ExportedSavedSites -> showExportedSavedSites(it.exportSavedSitesResult)
+                is OpenBookmarkFolder -> openBookmarkFolder(it.bookmarkFolder)
+                is ShowEditBookmarkFolder -> editBookmarkFolder(it.bookmarkFolder)
+                is DeleteBookmarkFolder -> deleteBookmarkFolder(it.bookmarkFolder)
+                is ConfirmDeleteBookmarkFolder -> confirmDeleteBookmarkFolder(it.bookmarkFolder)
+                is LaunchBookmarkImport -> launchBookmarkImport()
+                is ShowFaviconsPrompt -> showFaviconsPrompt()
             }
         }
     }
 
+    private fun showFaviconsPrompt() {
+        val faviconPrompt = FaviconPromptSheet.Builder(this)
```

Comment:
Wondering if this could be a custom view with its own viewmodel? similar to how we are implementing some view like NotifymeView, or the view we show in syncsettings.

Reply:
I thought about it, but we have the issue that this is not a `View` attached to the tree hierarchy until it’s visible because it’s a `BottomSheet`. Meaning, we can’t inject `ViewModels` using the `findViewTreeViewModelStoreOwner`.

Since this is only used in two places, I found it easier to treat it as a simple bottom sheet, and let the `ViewModels` of the screens where it’s shown to deal with the logic.